week1/plot_trips.R

```diff
@@ -1,62 +1,154 @@
-# (you can skip this and come back to it tomorrow if we haven't covered pivot_wider() yet)
-
-########################################
-# plot weather data
-########################################
-# plot the minimum temperature (on the y axis) over each day (on the x axis)
-
-# plot the minimum temperature and maximum temperature (on the y axis, with different colors) over each day (on the x axis)
-# hint: try using the pivot_longer() function for this to reshape things before plotting
-# (you can skip this and come back to it tomorrow if we haven't covered reshaping data yet)
-
-########################################
-# plot trip and weather data
-########################################
-
-# join trips and weather
-trips_with_weather <- inner_join(trips, weather, by="ymd")
-
-# plot the number of trips as a function of the minimum temperature, where each point represents a day
-# you'll need to summarize the trips and join to the weather data to do this
-
-# repeat this, splitting results by whether there was substantial precipitation or not
-# you'll need to decide what constitutes "substantial precipitation" and create a new T/F column to indicate this
-
-# add a smoothed fit on top of the previous plot, using geom_smooth
-
-# compute the average number of trips and standard deviation in number of trips by hour of the day
-# hint: use the hour() function from the lubridate package
-
-# plot the above
-
-# repeat this, but now split the results by day of the week (Monday, Tuesday, ...) or weekday vs. weekend days
-# hint: use the wday() function from the lubridate package
+########################################
+# load libraries
+########################################
+
+# load some packages that we'll need
+library(tidyverse)
+library(scales)
+library(lubridate)
+
+# be picky about white backgrounds on our plots
+theme_set(theme_bw())
+
+# load RData file output by load_trips.R
+load('trips.RData')
+
+
+########################################
+# plot trip data
+########################################
+
+# plot the distribution of trip times across all rides (compare a histogram vs. a density plot)
+ggplot(data = trips, mapping = aes(x = starttime)) +
+  geom_histogram()
+
+ggplot(trips, mapping = aes(x = starttime)) + 
+  geom_density(fill = "grey")
+
+
+# plot the distribution of trip times by rider type indicated using color and fill (compare a histogram vs. a density plot)
+ggplot(data = trips, mapping = aes(x = starttime, color = gender, fill = gender)) +
+  geom_histogram()
+
+ggplot(trips, mapping = aes(x = starttime)) + 
+  geom_density(mapping = aes(fill = gender))
+
+# plot the total number of trips on each day in the dataset
+ggplot(trips, mapping = aes(x = ymd)) +
+  geom_bar() +
+  scale_y_continuous(label = comma) 
+
+# plot the total number of trips (on the y axis) by age (on the x axis) and gender (indicated with color)
+trips %>% 
+  mutate(age = 2014 - birth_year) %>%

```

Comment:
Oh I see you did 2014 here.

I find it's easiest when you are putting in arbitrary numbers (like 2014, or the precipitation boundary) to create a variable and then call it in your processing lines so that the script is consistent everywhere. Totally not a big deal here since we use it twice but if you find yourself working with huge scripts with a bunch of arbitrary boundaries it can get messy quick.